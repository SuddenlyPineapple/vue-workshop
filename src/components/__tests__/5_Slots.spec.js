import { shallowMount } from "@vue/test-utils";
import Slots from "../5_Slots";

describe("5_Slots", () => {
  const baseComponent = {
    name: "base-component",
    template: "<v-btn>Yeah!</v-btn>"
  };

  const headerComponent = {
    name: "baz-component",
    template: "<h1>I'm heading!</h1>"
  };

  const getWrapper = (component1, component2) =>
    shallowMount(Slots, {
      slots: { default: component1, heading: component2 }
    });

  it("should render given component", () => {
    const wrapper = getWrapper(baseComponent, headerComponent);
    expect(wrapper).toMatchSnapshot();
  });
});
