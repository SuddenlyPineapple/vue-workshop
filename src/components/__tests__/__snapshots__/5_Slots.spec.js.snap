// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`5_Slots should render given component 1`] = `
<v-card color="primary" class="pa-5 white--text">
  <h1>I'm heading!</h1>
  <v-btn>Yeah!</v-btn>
</v-card>
`;
